moveit_ros/visualization/rviz_plugin_render_tools/src/trajectory_visualization.cpp

```diff
@@ -436,9 +440,8 @@ void TrajectoryVisualization::update(float wall_dt, float ros_dt)
         if (!loop_display_property_->getBool() && trajectory_slider_panel_)
           trajectory_slider_panel_->pauseButton(true);
       }
-      current_state_time_ = 0.0f;
+      current_state_time_ = current_state_time_ - tm;
     }
-    current_state_time_ += wall_dt;
```

Comment:
Why do you need to move the time-increment from here to an additional `else` branch above?
In both cases time is incremented when `animating_path_` is true.

Reply:
Previously the time increment was done after the state time check. As you pointed out, I moved it to the front. I did so because `wall_dt` given as a parameter corresponds to the delta since the last time `update()` was called. So it would be wrong to use this delta for the next check instead of the current one.
The additional `else` is needed to ensure that we don't increase the time directly after setting it to zero when a new display was started because in that case adding a delta wouldn't make any sense.